### PR TITLE
Add prolific_pid and start date to the admin panel

### PR DESCRIPTION
**Pull request by Era-Dorta**
Add the prolific PID and start date to the admin panel. This is added to the journeys like so:
Screenshot from 2024-04-05 15-47-22

### CODE REVIEW

File: covfee/client/admin/hit_block/journey_buttons.tsx [hunk 1]
```diff
@@ -30,6 +30,32 @@ export const JourneyRow = ({
 }: JourneyRowProps) => {
   const { addChats } = React.useContext(chatContext)
   const { getUrl } = useJourneyFns(journey)
+  const [annotator_data, setAnnotatorData] = React.useState<Annotator>(null)
```
Comment (Kenneth-Funes-TU-Delft) [inline covfee/client/admin/hit_block/journey_buttons.tsx:33]: to follow the coding standard in covfee, remove underscores and use camelCase. 
`const [annotatorData, setAnnotatorData] = React.useState<Annotator>(null)`

Comment (Kenneth-Funes-TU-Delft) [reply covfee/client/admin/hit_block/journey_buttons.tsx:33]: moreover, the variable `annotatorData` name may be simplified into simply `annotator`

File: covfee/client/admin/hit_block/journey_buttons.tsx [hunk 2]
```diff
@@ -30,6 +30,32 @@ export const JourneyRow = ({
...
+  const [show_annotator_data, setShowAnnotatorData] =
```
Comment (Kenneth-Funes-TU-Delft) [inline covfee/client/admin/hit_block/journey_buttons.tsx:34]: same, please use camelCase standard

File: covfee/client/admin/hit_block/journey_buttons.tsx [hunk 3]
```diff
@@ -47,16 +73,22 @@ export const JourneyRow = ({
...
             }}
           >
             <ApiOutlined />
           </span>
         </LinkContainer>
         <span> </span>
-        <span>{journey.id.substring(0, 10)}</span> <LinkOutlined />
+        <span>{journey.id.substring(0, 10)} </span> <LinkOutlined />
       </a>
+      {show_annotator_data && (
```
Comment (Kenneth-Funes-TU-Delft) [inline covfee/client/admin/hit_block/journey_buttons.tsx:86]: I think you can remove the state of `show_annotator_data` and simply use `annotator_data && ...` as they are redundant, i.e., show_annotator_data is true always when annotator_data is not null

File: covfee/client/models/Journey.ts [hunk 4]
```diff
@@ -85,3 +85,8 @@ export const submitJourney = (id: string) => {
   }
   return fetcher(url, requestOptions).then(throwBadResponse)
 }
+
+export const fetch_annotator_data = (id: string) => {
+  const url = Constants.api_url + "/journeys/" + id + "/annotator_data"
```
Comment (Kenneth-Funes-TU-Delft) [inline covfee/client/models/Journey.ts:90]: perhaps simplify the API into `const url = Constants.api_url + "/journeys/" + id + "/annotator"` . I don't know how informative the `_data` is

File: covfee/client/types/journey.ts [hunk 5]
```diff
@@ -22,4 +27,5 @@ export interface JourneyType extends Omit<JourneySpec, "nodes"> {
   num_connections: number
   curr_node_id: number
   status: JourneyApiStatus
+  annotator?: Annotator
```
Comment (Kenneth-Funes-TU-Delft) [inline covfee/client/types/journey.ts:30]: `annotator?: Annotator | null`

File: covfee/server/rest_api/journeys.py [hunk 6]
```diff
@@ -81,3 +81,12 @@ def node_ready(jid, nidx, value):
...
 
     return "", 200
+
+# Return the annotator_data for the annotator working on this journey
+@api.route("/journeys/<jid>/annotator_data")
+@admin_required
+def annotator_data(jid):
+    res :JourneyInstance  = app.session.query(JourneyInstance).get(bytes.fromhex(jid))
+    if res.annotator is None:
+        return {}
```
Comment (Kenneth-Funes-TU-Delft) [inline covfee/server/rest_api/journeys.py:91]: does `return {}` creates a `null` payload in the frontend ?

**Review by Kenneth-Funes-TU-Delft — CHANGES_REQUESTED** [review]
Looks very good. I made a few recommendations mostly for consistency with the current implementation and small simplications.

File: covfee/client/admin/hit_block/journey_buttons.tsx (hunk 1 above)
Comment (Era-Dorta) [reply covfee/client/admin/hit_block/journey_buttons.tsx:33]: Fixed in 06bbf53eba006fb394fef90cc275f8062d7a3762

File: covfee/client/admin/hit_block/journey_buttons.tsx (hunk 2 above)
Comment (Era-Dorta) [reply covfee/client/admin/hit_block/journey_buttons.tsx]: Fixed in 06bbf53eba006fb394fef90cc275f8062d7a3762

File: covfee/client/admin/hit_block/journey_buttons.tsx (hunk 3 above)
Comment (Era-Dorta) [reply covfee/client/admin/hit_block/journey_buttons.tsx]: Neat! Fixed in 06bbf53eba006fb394fef90cc275f8062d7a3762

File: covfee/server/rest_api/journeys.py (hunk 6 above)
Comment (Era-Dorta) [reply covfee/server/rest_api/journeys.py:91]: It shows as `{}` in typescript, how do I check for that?

File: covfee/client/models/Journey.ts (hunk 4 above)
Comment (Era-Dorta) [reply covfee/client/models/Journey.ts]: Fixed in 9fbd57d

File: covfee/server/rest_api/journeys.py (hunk 6 above)
Comment (Era-Dorta) [reply covfee/server/rest_api/journeys.py:91]: I tried with ce37882 , not sure if it makes sense. I tried `=== null` and `=== undefined` but those didn't work.

**Review by Kenneth-Funes-TU-Delft — APPROVED** [review]
Looks very good. Nice solution.

File: covfee/client/types/journey.ts (hunk 5 above)
Comment (Era-Dorta) [reply covfee/client/types/journey.ts:30]: Delete this attribute

File: covfee/server/rest_api/journeys.py (hunk 6 above)
Comment (Era-Dorta) [reply covfee/server/rest_api/journeys.py:91]: Remove the checks for the prolific and date attributes

**Pull request merged**